framework/test/jest/unit/specs/modules/chain/synchronizer/block_synchronization_mechanism/block_synchronization_mechanism.spec.js

```diff
@@ -14,130 +14,236 @@
 
 'use strict';
 
-const BlockSynchronizationMechanism = require('../../../../../../../../src/modules/chain/synchronizer/block_synchronization_mechanism');
+const {
+	BlockSynchronizationMechanism,
+} = require('../../../../../../../../src/modules/chain/synchronizer/block_synchronization_mechanism');
 const peersList = require('./peers');
+const {
+	Block: blockFixture,
+} = require('../../../../../../../mocha/fixtures/blocks');
 
-describe('BlockSynchronizationMechanism', () => {
-	const stubs = {
-		channel: {
-			invoke: jest.fn(),
-		},
-		modules: {
+describe('block_synchronization_mechanism', () => {
+	afterEach(async () => {
+		jest.clearAllMocks();
```

Comment:
isn't this automatic? @yatki

Reply:
It not clearing the mocks automatically, so I have to add these lines.

Reply:
@nazarhussain, you can avoid using this call by simply moving mock definitions into beforeEach statements or adding `clearMocks: true` option to jest.config.js. 

Notice the assignment below is in `describe` block. and jest.fn() creates a mock function only once. If you move this assignment into beforeEach you won't need to call `jest.clearAllMocks` because in each hook they'll be reassigned.

```js
	const slotsMock = {
			getSlotNumber: jest.fn(),
		};
```
@shuse2 @limiaspasdaniel I think this also answers your comment above.

Reply:
So that's actually reassigning the mock objects completely, not clearing old mock values and states. Which I felt completely wrong in reference to clearing mocks. I don't feel its a good idea to create variables with `let` and then assigning their mocks in `beforeEach`. Using `const` with definition and clearing mocks after each is right way.

Reply:
@shuse2 @yatki Adding `clearMocks: true` in the jest config works for these tests then 7 other tests starts failing. So let's add this config and remove `clearAllMocks` as separate issue.

Reply:
Here is the issue. https://github.com/LiskHQ/lisk-sdk/issues/4026